scheduler/src/managers/workload/manager.rs

```diff
@@ -0,0 +1,60 @@
+use crate::grpc::clients::send_workload_signal;
+use crate::managers::node_agent::manager::NodeAgentManager;
+use orka_proto::node_agent::workload_signal::Signal;
+use orka_proto::scheduler_controller::Empty;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use tonic::{Response, Status};
+
+pub struct WorkloadManager {
+    /// The managed instances.
+    /// Stored under the form `instance_id -> agent_id`.
+    instances: HashMap<String, String>,
+}
+
+impl WorkloadManager {
+    /// Creates a new workload manager
+    pub fn new() -> Self {
+        WorkloadManager {
+            instances: HashMap::new(),
+        }
+    }
+
+    /// Add a new instance to the list
+    pub fn add_instance(&mut self, instance_id: &String, agent_id: String) {
+        self.instances.insert(instance_id.to_string(), agent_id);
+    }
+
+    pub fn find_related_agent(&self, instance_id: &String) -> Option<&String> {
```

Comment:
Take a string slice instead.
```suggestion
    pub fn find_related_agent(&self, instance_id: &str) -> Option<&String> {
```